src/main/scala/com/kinja/amqp/AmqpProducer.scala

```diff
@@ -50,7 +50,7 @@ class AmqpProducer(
 		channel ? Publish(exchange.name, routingKey, bytes, properties = Some(properties), mandatory = true, immediate = false) map {
 			case Ok(_, Some(MessageUniqueKey(deliveryTag, channelId))) => {
 				messageStore.saveMessage(
-					Message(None, routingKey, exchange.name, json.toString, Some(channelId), Some(deliveryTag), saveTimeMillis)
+					Message(None, routingKey, exchange.name, json.toString, Some(channelId.toString), Some(deliveryTag), saveTimeMillis)
```

Comment:
If we convert it to string everywhere we use it we should just convert it it to string right after we generated it. I have no problems with the use of UUID either, the actual implementation can deal with it (mysql can use it, redis can convert. it's always cheaper on the app server side as there will be always one master mysql/redis, and probably a lot more app servers).

I would go with strings only, but I'm okay with leaving it as UUID. Just don't convert it on the interface level then.